src/main/java/com/marklogic/kafka/connect/source/MarkLogicSourceConfig.java

```diff
@@ -29,6 +29,9 @@ public class MarkLogicSourceConfig extends MarkLogicConfig {
     enum OUTPUT_TYPE {JSON, XML, CSV}
     private static final CustomRecommenderAndValidator OUTPUT_FORMAT_RV =
         new CustomRecommenderAndValidator(Arrays.stream(MarkLogicSourceConfig.OUTPUT_TYPE.values()).map(Enum::toString).toArray(String[]::new));
+    public static final String KEY_STRATEGY = "ml.source.key.strategy";
+    private static final CustomRecommenderAndValidator KEY_STRATEGY_RV =
+        new CustomRecommenderAndValidator("UUID", "TIMESTAMP", "NULL");
```

Comment:
I think that if "NULL" is going to be an actual strategy, then we should have an implementation for it.
Also, I wonder if we can come up with a better name, so that "NULL" isn't overloaded.

Reply:
I like this - a `NullKeyGenerator` makes sense and means that whatever calls the `KeyGenerator` doesn't need to know about the concept of `null` as a key.

I'm not sure though that "NULL" should be a choice for the user. I feel like making the field optional works, and the absence of a user choice means we use `NullKeyGenerator`. Though - I can see the benefit of making this explicit, but I wonder if "None" would be a little less developer-speakish.

Reply:
Yep, I think "None" is good.